lec_8_HW/task_2.js

```diff
@@ -0,0 +1,179 @@
+// console.log(enterprises);
+
+// 4. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function addDepartment(enterpriseId, departmentName) {
+    const enterprise = enterprises.find(enterprise => enterprise.id === enterpriseId);
+    if (enterprise) {
+        const newId =
+            enterprises.flatMap(e => e.departments).reduce((maxId, department) => Math.max(maxId, department.id), 0) +
+            1;
+        enterprise.departments.push({
+            id: newId,
+            name: departmentName,
+            employees_count: 0,
+        });
+    }
+}
+
+// addDepartment(1, "NewDepartment_1");
+// console.log(enterprises);
+
+// 5. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function editEnterprise(enterpriseId, newName) {
+    const enterprise = enterprises.find(enterprise => enterprise.id === enterpriseId);
+    if (enterprise) {
+        enterprise.name = newName;
+    }
+}
+
+// editEnterprise(1, "NewNameOfEnterprise");
+// console.log(enterprises);
+
+// 6. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function editDepartment(departmentId, newName) {
+    const department = enterprises.flatMap(e => e.departments).find(department => department.id === departmentId);
+    if (department) {
+        department.name = newName;
+    }
+}
+
+// editDepartment(7, "NewNameOfDepartment");
+// console.log(enterprises);
+
+// 7. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function deleteEnterprise(enterpriseId) {
+    const index = enterprises.findIndex(enterprise => enterprise.id === enterpriseId);
+    if (index !== -1) {
+        enterprises.splice(index, 1);
+    }
+}
+
+// deleteEnterprise(9);
+// console.log(enterprises);
+
+// 8. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function deleteDepartment(departmentId) {
+    for (const enterprise of enterprises) {
+        const index = enterprise.departments.findIndex(
+            department => department.id === departmentId && department.employees_count === 0,
+        );
+        if (index !== -1) {
+            enterprise.departments.splice(index, 1);
+            break;
+        }
+    }
+}
+
+// deleteDepartment(10);
+// console.log(enterprises);
+
+// 9. ///////////////////////////////////////////////////////////////////////////////////////////////////////////////
+function moveEmployees(fromDepartmentId, toDepartmentId) {
+    const fromDepartment = enterprises
+        .flatMap(e => e.departments)
+        .find(department => department.id === fromDepartmentId);
+    const toDepartment = enterprises.flatMap(e => e.departments).find(department => department.id === toDepartmentId);
+
+    if (fromDepartment && toDepartment) {
```

Comment:
Add check whether two departments belong to the same enterprise

Reply:
accepted, code corrected